src/Riok.Mapperly/Configuration/AttributeDataAccessor.cs

```diff
@@ -161,13 +163,20 @@ is InvocationExpressionSyntax
         throw new InvalidOperationException($"Cannot create {nameof(StringMemberPath)} from {arg.Kind}");
     }
 
-    private static object?[] BuildArrayValue(TypedConstant arg, Type targetType)
+    private static IReadOnlyCollection<object?> BuildArrayValue(TypedConstant arg, Type targetType)
     {
         if (!targetType.IsGenericType || targetType.GetGenericTypeDefinition() != typeof(IReadOnlyCollection<>))
             throw new InvalidOperationException($"{nameof(IReadOnlyCollection<object>)} is the only supported array type");
 
         var elementTargetType = targetType.GetGenericArguments()[0];
-        return arg.Values.Select(x => BuildArgumentValue(x, elementTargetType, null)).ToArray();
+        var listType = typeof(List<>).MakeGenericType(elementTargetType);
+        var list = (IList)Activator.CreateInstance(listType, arg.Values.Length);
```

Comment:
FYI: All of this does not seem to be covered by tests

Reply:
Reverted.